bit_matrix.go

```diff
@@ -0,0 +1,98 @@
+package rsmt2d
+
+import "fmt"
+
+type bitMatrix struct {
+	mask       []uint64
+	squareSize int
+}
+
+func newBitMatrix(squareSize int) bitMatrix {
+	bits := squareSize * squareSize
+	return bitMatrix{mask: make([]uint64, (bits+63)/64), squareSize: squareSize}
+}
+
+// i = rowIndex*squareSize+colIdx
+func (bm bitMatrix) SetFlat(i int) {
+	bm.mask[i/64] |= uint64(1) << uint(i%64)
+}
+
+func (bm bitMatrix) Get(row, col int) bool {
```

Comment:
Do these methods have to be exported?

Reply:
Not really, no. Everything is in a single flat root package here anyways and even the type itself is not exported `bitMatrix`. Both the type and these methods would need to be exported if this was moved into a separaste go package. 
I only exported all these methods so it is clear which are meant to be used as an API, in other words, to communicate the surface of this if it was a library. The caller can currently bypass that anyways: they could directly mingle with the underlying `mask` field.

Reply:
I can definitely dig that, nice

Reply:
IMO, we should put stuff like this into an internal package: https://dave.cheney.net/2019/10/06/use-internal-packages-to-reduce-your-public-api-surface
But maybe we can tackle this together with #12 later.